build.cake

```diff
@@ -353,75 +362,98 @@ Task("Restore")
-
-        RunTool(env.DotNetCommand, "build", folder)
-            .ExceptionOnError($"Failed to build '{folder}'.");
-    }
-
-    var platform = Platform.Current;
-
-    // Restore and build Windows-only test assets
-    if (platform.IsWindows)
-    {
-        foreach (var project in buildPlan.WindowsOnlyTestAssets)
+        DotNetCoreRestore(new DotNetCoreRestoreSettings()
         {
-            Information("Restoring and building: {0} (windows-only)...", project);
+            ToolPath = env.DotNetCommand,
+            WorkingDirectory = folder
+            });
+        DotNetCoreBuild(folder, new DotNetCoreBuildSettings()
+        {
+            ToolPath = env.DotNetCommand,
+            WorkingDirectory = folder
+        });
+    });
 
-            var folder = CombinePaths(env.Folders.TestAssets, "test-projects", project);
+Task("PrepareTestAssets:WindowsTestAssets")
+    .WithCriteria(Platform.Current.IsWindows)
+    .IsDependeeOf("PrepareTestAssets")
+    .DoesForEach(buildPlan.WindowsOnlyTestAssets, (project) =>
+    {
+        Information("Restoring and building: {0}...", project);
 
-            RunTool(env.DotNetCommand, "restore", folder)
-                .ExceptionOnError($"Failed to restore '{folder}'.");
+        var folder = CombinePaths(env.Folders.TestAssets, "test-projects", project);
 
-            RunTool(env.DotNetCommand, "build", folder)
-                .ExceptionOnError($"Failed to build '{folder}'.");
-        }
-    }
+        DotNetCoreRestore(new DotNetCoreRestoreSettings()
+        {
+            ToolPath = env.DotNetCommand,
+            WorkingDirectory = folder
+        });
+        DotNetCoreBuild(folder, new DotNetCoreBuildSettings()
+        {
+            ToolPath = env.DotNetCommand,
+            WorkingDirectory = folder
+        });
+    });
 
-    if (AllowLegacyTests())
+Task("PrepareTestAssets:LegacyTestAssets")
+    .WithCriteria(() => AllowLegacyTests())
+    .IsDependeeOf("PrepareTestAssets")
+    .DoesForEach(buildPlan.LegacyTestAssets, (project) =>
     {
+        var platform = Platform.Current;
         if (platform.IsMacOS && platform.Version.Major == 10 && platform.Version.Minor == 12)
         {
             // Trick to allow older .NET Core SDK to run on Sierra.
             Environment.SetEnvironmentVariable("DOTNET_RUNTIME_ID", "osx.10.11-x64");
         }
 
-        // Restore and build legacy test assets with legacy .NET Core SDK
-        foreach (var project in buildPlan.LegacyTestAssets)
-        {
-            Information("Restoring and building project.json: {0}...", project);
+        // RunTool(env.LegacyDotNetCommand, "restore", folder)
+        //     .ExceptionOnError($"Failed to restore '{folder}'.");
 
-            var folder = CombinePaths(env.Folders.TestAssets, "test-projects", project);
+        // RunTool(env.LegacyDotNetCommand, $"build", folder)
+        //     .ExceptionOnError($"Failed to build '{folder}'.");
```

Comment:
Did you mean to leave these here commented?

Reply:
Nope, they were for reference to make sure I had made the correct calls.  Fixing.